server/src/routes/federation.rs

```diff
@@ -1,18 +1,35 @@
 use super::*;
-use crate::apub;
+use crate::apub::community::*;
+use crate::apub::community_inbox::community_inbox;
+use crate::apub::post::get_apub_post;
+use crate::apub::user::*;
+use crate::apub::user_inbox::user_inbox;
+use crate::apub::APUB_JSON_CONTENT_TYPE;
 
 pub fn config(cfg: &mut web::ServiceConfig) {
-  cfg
-    .route(
-      "/federation/c/{community_name}",
-      web::get().to(apub::community::get_apub_community),
-    )
-    .route(
-      "/federation/c/{community_name}/followers",
-      web::get().to(apub::community::get_apub_community_followers),
-    )
-    .route(
-      "/federation/u/{user_name}",
-      web::get().to(apub::user::get_apub_user),
-    );
+  if Settings::get().federation.enabled {
+    println!("federation enabled, host is {}", Settings::get().hostname);
+    cfg
+      .service(
+        web::scope("/")
+          .guard(guard::Header("Accept", APUB_JSON_CONTENT_TYPE))
```

Comment:
you may not want to _only_ handle application/activity+json. Some clients might request application/ld+json, application/json, */*, or nothing at all

Reply:
it's probably valid to ignore the \*/\* and nothing cases, but you should definitely consider supporting other json-type accept headers